Jenkinsfile

```diff
@@ -36,6 +36,12 @@ def doTest(env, target = "test") {
   junit '**/build/test-results/**/TEST-*.xml'
 }
 
+def runTestOnDevBranch(env) {
+  if (!isChangeRequest(env)) {
+    doTest(env)
+  }
+}
```

Comment:
If they don't run on PRs, I don't think anyone will validate the results for the branch builders. We should probably remove Java 11 and 17 altogether now that we have ported most tests to run with Java 16 and higher (previously many were excluded).

I had suggested this in an internal conversation, so good to see similar action here.

Reply:
What do you think about leaving the 11 and 17 branch builders as-is until 4.0? This will need some rework then to effect the KIPs that are changing the minimum version requirements and having different minimums for the different modules. We can figure out a strategy for branch & pr builds then.

Reply:
I don't see the value - once we drop support for Java 8, we will simply change the Java 8 builder to Java 11. Am I missing something?

Reply:
Ah, yes - we'll also have to configure the brokers, etc. to use Java 17. In any case, I think whatever we have to do there won't be any harder if we just delete the Java 11 and 17 builders for now. And it will make the 3.8 builds lighter.

Reply:
I'm hesitant to sign up future release managers for additional verification of the 11 and 17 builds that the CI has been doing for previous releases. I don't expect any failures to be introduced that affect 11 and 17 but not 8 or 21, but I would rather find those failures out in CI after merge than during the RC generation stage.

And for selfish reasons, I also would like to continue seeing the 11 and 17 trunk results published so I can do flaky test analysis. With this patch as-is, we're keeping the same number of test runs on trunk, but if we removed these entirely we would be cutting those in half. I don't pay attention to the PR builds for statistics because contributors may have bad patches applied that aren't worth investigating.